### PR TITLE
Disable autobump by default

### PR DESCRIPTION
**Pull request by sisuresh**
# Description

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/transactions/TransactionFrame.cpp [hunk 1]
```diff
@@ -1685,7 +1685,8 @@ TransactionFrame::applyExpirationBumps(Application& app, AbstractLedgerTxn& ltx)
         if (lte && isSorobanDataEntry(lte.current().data))
         {
             // Must enforce minimum expirations on write
-            bump(lte.current(), autoBumpEnabled(lte.current()), true);
+            bump(lte.current(),
+                 autoBumpEnabled(lte.current()) && autoBumpLedgers > 0, true);
```
Comment (dmkozh) [inline src/transactions/TransactionFrame.cpp:1689]: nit: swap the conditions to not touch lte when autobump is disabled (same below)?

Comment (sisuresh) [reply src/transactions/TransactionFrame.cpp]: Done. I also updated the read section again because we would still `load` and update lastModified.

**Comment by dmkozh** [comment]
r+ d0de504

**Comment by dmkozh** [comment]
r+ bf048c0

**Pull request merged**